reference/halyard/component-sizing.md

```diff
@@ -0,0 +1,52 @@
+---
+layout: single
+title:  "Component Sizing"
+sidebar:
+  nav: reference
+---
+
+{% include toc %}
+
+
+Within the halconfig custom sizing can be specified for each Spinnaker component. Custom sizing must be manually added to the halconfig, there are no CLI commands or flags to enable these settings. This feature is currently only supported for distributed deployments using Kubernetes (`hal config deploy edit --type distributed --account-name my-k8s-cluster`).
+
+## Kubernetes
+
+Container requests and limits for cpu and memory can be specified in the customSizing section of the halconfig. Note that these sizings will also be applied to any sidecars running along side the container as well. Please see the example for details.
+
+### Example
+```
+deploymentEnvironment:
+  customSizing:
+    spin-clouddriver:
+      limits:
+        cpu: 1
+        memory: 1Gi
+      requests:
+        cpu: 250m
+        memory: 512Mi
+    spin-echo:
+      limits:
+        cpu: 250m
+        memory: 512Mi
+      requests:
+        cpu: 100m
+        memory: 128Mi
+```
+
+Limits and requests follow the Kubernetes conventions [documented here](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/).
+
+### Updating JAVA_OPTS
```

Comment:
Is this section necessary when potentially using `JAVA_DOCKER_MEMORY_OPTS='-XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap -XX:MaxRAMFraction=1 -XshowSettings:vm'` described by @lwander @greenkiwi [here](https://github.com/spinnaker/halyard/issues/683#issuecomment-329791497)?

Reply:
@rosskevin yes, that setting only works on a newer version of Java than all the microservices are using at the moment.

background here: https://github.com/spinnaker/halyard/pull/724#issuecomment-335577794

Reply:
Related: https://github.com/spinnaker/spinnaker/issues/2058